File: test/sync_call.jl
#!/usr/bin/env julia

include("../src/MsgPackRpcClient.jl")
include("dataset.jl")
include("const.jl")
using MsgPackRpcClient
using Base.Test

function test_data(data, number = 1)
  for d in data
    println("$(number): $d ")
    try
      result = MsgPackRpcClient.call(session, d["method"], d["arg"]; sync = true)
      #println("d[\"arg\"] = ", d["arg"], ", result = $result", ", first(result]) = ", first(result))
      if haskey(d, "expect")
        @test first(result) == d["expect"]
      else
        @test first(result) == d["arg"]
      end
      println("  => OK")
    catch
      println("Error on d[\"arg\"] = ", d["arg"])
    end
    number += 1
  end
end

session = MsgPackRpcClientSession.create()

test_data(get_dataset())

session.destroy()

File: test/sync_call.rb
#!/usr/bin/env ruby
# -*- coding: utf-8 -*-

require 'msgpack/rpc'
require 'test/unit'
require './dataset'
require './const'

class SyncCall < Test::Unit::TestCase

  def get_session
    session = MessagePack::RPC::Client.new(HOSTNAME, PORT)
    session.timeout = TIMEOUT
    session
  end

  def test_sync_call
    dataset = DataSet.new.dataset
    session = get_session
    dataset.each do |d|
      result = session.call(d["method"], d["arg"])
      p result
      assert_equal(d["arg"], result.first)
    end
    session.close()
  end
end

